From 5144b1c7b3518863905137af04e75392e9670bb8 Mon Sep 17 00:00:00 2001
From: Sean McBride <sean@rogue-research.com>
Date: Sat, 3 Feb 2024 22:19:51 -0500
Subject: [PATCH] descriptor: Restore implicitly casted-away const

---
 libusb/descriptor.c   | 4 ++--
 libusb/version_nano.h | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/libusb/descriptor.c b/libusb/descriptor.c
index df670ee29..45056b12f 100644
--- a/libusb/descriptor.c
+++ b/libusb/descriptor.c
@@ -694,14 +694,14 @@ int API_EXPORTED libusb_get_ss_endpoint_companion_descriptor(
 	const struct libusb_endpoint_descriptor *endpoint,
 	struct libusb_ss_endpoint_companion_descriptor **ep_comp)
 {
-	struct usbi_descriptor_header *header;
+	const struct usbi_descriptor_header *header;
 	const uint8_t *buffer = endpoint->extra;
 	int size = endpoint->extra_length;
 
 	*ep_comp = NULL;
 
 	while (size >= DESC_HEADER_LENGTH) {
-		header = (struct usbi_descriptor_header *)buffer;
+		header = (const struct usbi_descriptor_header *)buffer;
 		if (header->bDescriptorType != LIBUSB_DT_SS_ENDPOINT_COMPANION) {
 			if (header->bLength < DESC_HEADER_LENGTH) {
 				usbi_err(ctx, "invalid descriptor length %u",
diff --git a/libusb/version_nano.h b/libusb/version_nano.h
index deb992fe7..3e99f0837 100644
--- a/libusb/version_nano.h
+++ b/libusb/version_nano.h
@@ -1 +1 @@
-#define LIBUSB_NANO 11908
+#define LIBUSB_NANO 11909
